src/identity_policy_provider.py

```diff
@@ -0,0 +1,198 @@
+
+
+request_schema = {
+    "type": "object",
+    "required": ["Identity", "PolicyName", "PolicyDocument"],
+    "properties": {
+        "Identity": {
+            "type": "string",
+            "description": "that the policy will apply to"
+        },
+        "PolicyName": {
+            "type": "string",
+            "description": "of the policy"
+        },
+        "PolicyDocument": {
+            "type": "object",
+            "description": "the permissions for the principal",
+            "required": ["Statement"],
+            "properties": {
+                "Statement": {
+                    "type": "array",
+                    "items": {
+                        "type": "object",
+                        "required": ["Effect", "Principal", "Action"],
+                        "properties": {
+                            "Effect": {
+                                "type": "string",
+                                "enum": ["Allow", "Deny"]
+                            },
+                            "Principal": {
+                                "type": "object",
+                                "oneOf": [{
+                                    "required": ["AWS"],
+                                    "properties": {
+                                        "AWS": {
+                                            "type": "string"
+                                        }
+                                    }
+                                }, {
+                                    "required": ["AWS"],
+                                    "properties": {
+                                        "AWS": {
+                                            "type": "array",
+                                            "items": {
+                                                "type": "string"
+                                            }
+                                        }
+                                    }
+                                }]
+                            },
+                            "Action": {
+                                "type": "array",
+                                "items": {
+                                    "type": "string",
+                                    "enum": ["ses:SendEmail", "ses:SendRawEmail"]
+                                }
+                            }
+                        }
+                    }
+                },
+                "Version": {
+                    "type": "string",
+                    "description": "of the policy document"
+                }
+            }
+        }
+    }
+}
+
+
+class IdentityPolicyProvider(ResourceProvider):
+
+    def __init__(self):
+        super(IdentityPolicyProvider, self).__init__()
+        self.request_schema = request_schema
+        self.ses = boto3.client('ses')
+
+    def convert_property_types(self):
+        self.heuristic_convert_property_types(self.properties)
+
+    def create(self):
+        self.upsert()
```

Comment:
If a identity policy with the same name already exist, the create should error out. Otherwise you can unintentionally hi-jack somebody else's policy.

Reply:
I pushed some fixes:
* Duplicate policy names will throw an error
* Changing the policy name will also throw an error (this is also not possible in the console)
* Improved the policy name validation